### PR TITLE
Updates to use gtk2-common-themes and core18 base, fixes gtk themes

### PR DESCRIPTION
**Pull request by kenvandine**

### CODE REVIEW

File: snapcraft.yaml [hunk 1]
```diff
@@ -30,29 +43,39 @@ apps:
...
+    stage-packages:
+      - libxkbcommon0  # XKB_CONFIG_ROOT
+      - shared-mime-info
+      - libgtk2.0-0
+      - libgdk-pixbuf2.0-0
+      - libglib2.0-bin
+      - libgtk2.0-bin
+      - unity-gtk2-module
+      - locales-all
+      - libappindicator1
```
Comment (TingPing) [inline snapcraft.yaml]: Seems like a random package we don't need?

Comment (kenvandine) [reply snapcraft.yaml]: Those are for unity7, but it could work without those.

Comment (kenvandine) [reply snapcraft.yaml]: I dropped libappindicator and unity-gtk2-module as well as cleaned up some other things

**Comment by TingPing** [comment]
Since you removed the wrapper have you verified Lua actually works?

**Comment by kenvandine** [comment]
You're right, I did break Lua.  I'll fix that.

File: snapcraft.yaml [hunk 2]
```diff
@@ -5,54 +5,74 @@ description: >
...
+      - libgdk-pixbuf2.0-0
+      - libglib2.0-bin
+      - libgtk2.0-bin
+      - locales-all
+      - xdg-user-dirs
+        
   hexchat:
 # FIXME: Spell check needs dictionaries
 # FIXME: Notifications can't see the desktop file or icon
 # FIXME: Platform see above:
```
Comment (TingPing) [inline snapcraft.yaml:69]: So this still isn't using a "platform" and everything is still bundled; Is that doable yet?

File: snapcraft.yaml [hunk 3]
```diff
@@ -71,6 +91,10 @@ parts:
       - python3-dev
       - libperl-dev
       - libluajit-5.1-dev
+    stage-packages:
```
Comment (TingPing) [inline snapcraft.yaml:98]: Curious why these need to be added?

File: snapcraft.yaml (hunk 2 above)
Comment (ogra1) [reply snapcraft.yaml:69]: it uses the themeing (engines, themes and icons) platform snap, not the gtk libs themselves though

File: snapcraft.yaml (hunk 3 above)
Comment (ogra1) [reply snapcraft.yaml:98]: snapcraft stopped to blindly add stage-packages for all the -dev libs defined in build-packages to allow more fine grained contol about what is shipped....

**Comment by ogra1** [comment]
@kenvandine since the extremely ugly UI that you get with latest desktop bothered me i threw together a patch with the missing bits:

https://paste.ubuntu.com/p/fTQzMfZ4mh/

would be great if this could be integrated in this PR and finally land so the app looks sane and properly themed with new desktops again.

**Comment by kenvandine** [comment]
> @kenvandine since the extremely ugly UI that you get with latest desktop bothered me i threw together a patch with the missing bits:
> 
> https://paste.ubuntu.com/p/fTQzMfZ4mh/
> 
> would be great if this could be integrated in this PR and finally land so the app looks sane and properly themed with new desktops again.

Thanks!  I've integrated that in my branch.

**Comment by kenvandine** [comment]
@TingPing Sorry I never followed up on this.  I've been happily using my locally built snap :)

It would be great to get this merged.

**Pull request merged**